extensions/wikia/ArticleVideo/bower_components/html5-skin/js/controller.js

```diff
@@ -0,0 +1,1496 @@
+          this.mb.publish(OO.EVENTS.PLAY);
+        }.bind(this)
+      );
+      this.state.pluginsClickElement.mouseover(
+        function() {
+          this.showControlBar();
+          this.renderSkin();
+          this.startHideControlBarTimer();
+        }.bind(this)
+      );
+      this.state.pluginsClickElement.mouseout(
+        function() {
+          this.hideControlBar();
+        }.bind(this)
+      );
+      this.mb.publish(OO.EVENTS.UI_READY, {
+        videoWrapperClass: "innerWrapper",
+        pluginsClass: "oo-player-skin-plugins"
+      });
+      this.externalPluginSubscription();
+    },
+
+    onBitrateInfoAvailable: function(event, bitrates) {
+      if (bitrates && bitrates.bitrates) {
+        // copy array
+        var bitratesList = bitrates.bitrates.slice();
+				bitratesList.sort(function (a, b) {
+					return b.height - a.height;
+				});
+        this.state.videoQualityOptions.availableBitrates = bitratesList;
+        this.renderSkin({
+          "videoQualityOptions": {
+            "availableBitrates": bitratesList,
+            "selectedBitrate": this.state.videoQualityOptions.selectedBitrate
+          }
+        });
+      }
+    },
+
+    onClosedCaptionsInfoAvailable: function(event, info) {
+      if (!info || !info.videoId || !info.languages) {
+        return;
+      }
+      // Store info in cache in order to be able to restore it
+      // if this video element looses and then regains focus (like when an ad plays)
+      this.state.closedCaptionsInfoCache[info.videoId] = info;
+      this.setClosedCaptionsInfo(info.videoId);
+    },
+
+    onClosedCaptionCueChanged: function(event, data) {
+      if (data && data.length > 0) {
+        this.state.closedCaptionOptions.cueText = data;
+      } else {
+        this.state.closedCaptionOptions.cueText = null;
+      }
+      this.renderSkin();
+    },
+
+    onRelatedVideosFetched: function(event, relatedVideos) {
+      OO.log("onRelatedVideosFetched is called");
+      if (relatedVideos.videos) {
+        this.state.discoveryData = {relatedVideos: relatedVideos.videos};
+        this.state.upNextInfo.upNextData = relatedVideos.videos[0];
+        this.renderSkin();
+      }
+    },
+
+    // check if fullscreen is supported natively, set flag, add event listener for change
+    enableFullScreen: function() {
+      if (Fullscreen.enabled) {
+        this.state.isFullScreenSupported = true;
+        document.addEventListener(Fullscreen.raw.fullscreenchange, this.onFullscreenChanged.bind(this));
+      }
+    },
+
+    // iOS webkitSupportsFullscreen property is not valid until metadata has loaded
+    // https://developer.apple.com/library/safari/documentation/AudioVideo/Conceptual/Using_HTML5_Audio_Video/ControllingMediaWithJavaScript/ControllingMediaWithJavaScript.html#//apple_ref/doc/uid/TP40009523-CH3-SW13
+    enableIosFullScreen: function() {
+      if(!this.state.isFullScreenSupported) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the nested one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACollapsibleIfStatements)